shell/artifacts/Common/source/Multiplexer.js

```diff
@@ -37,40 +37,46 @@ defineParticle(({TransformationDomParticle}) => {
           this._connByHostedConn.set(hostedOtherConnection.name, connectionName);
         }
       }
+      this.setState({arc, type: views.get('list').type, hostedParticle, otherMappedViews, otherConnections});
 
-      this.on(views, 'list', 'change', async e => {
-        let listHandle = views.get('list');
-        let list = await listHandle.toList();
+      super.setViews(views);
+    }
 
-        if (list.length > 0) {
-          this.relevance = 0.1;
+    async willReceiveProps({list}) {
+      if (list.length > 0) {
+        this.relevance = 0.1;
+      }
+      let arc = this._state.arc;
```

Comment:
`state` is also a parameter (`willReceiveProps(props, state, oldProps, oldState)` like all of the lifecycle methods). Suggest you gather these variables via destructuring the parameter (either directly in the signature, or via a `const` declaration [like `const {arc, type, ...} = state;`]).

Fwiw, anytime you use `this._state` or `this._props`, it's a red-flag that something is awry.

Reply:
done